### PR TITLE
AP-3668: Update Client Undertaking question

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3668)

Ensure the providers enter a response for the client undertakings question regardless of whether they answer yes or no
This extrapolates and interpolates the values into split inputs on the page and merges into the single, additional_information, field in the database 

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/forms/providers/application_merits_task/client_offered_undertakings_form.rb [hunk 1]
```diff
@@ -3,28 +3,43 @@ module ApplicationMeritsTask
...
+      validates :additional_information_false, presence: true, unless: :requires_additional_information?
+
+      def initialize(*args)
+        super
+        extrapolate_additional_information
+      end
 
       def save
-        additional_information&.clear if offered?
+        attributes[:additional_information] = requires_additional_information? ? additional_information_true : additional_information_false
```
Comment (cpjmcquillan) [inline app/forms/providers/application_merits_task/client_offered_undertakings_form.rb:18]: What's the benefit of us storing the answer to two separate questions in the same place?

It seems like we're adding a fair amount of complexity when we could use specific columns in the database for each question.

Does this simplify things significantly elsewhere in the application?

**Review by cpjmcquillan — APPROVED** [review]
I added a question about the general approach, because I'm unsure what we gain from it.

But if we've really thought about it and decided this is worth it, then that's fine with me 👍

File: app/forms/providers/application_merits_task/client_offered_undertakings_form.rb (hunk 1 above)
Comment (colinbruce) [reply app/forms/providers/application_merits_task/client_offered_undertakings_form.rb:18]: It's a weird circumstance, the data provided is always additional information about the client's undertakings.
The reason for two questions is so we can provide a different prompt for the provider, roughly speaking. `What are their undertakings?` and `Why have they not provided any undertakings?`

Comment (colinbruce) [reply app/forms/providers/application_merits_task/client_offered_undertakings_form.rb:18]: I picked this pattern as it is already in use on the Domestic Abuse Summary page for police notification

**Pull request merged**